digiwf-integrations/digiwf-ticket-integration/digiwf-ticket-integration-core/src/main/java/de/muenchen/oss/digiwf/ticket/integration/adapter/out/s3/S3Adapter.java

```diff
@@ -51,6 +55,62 @@ public List<FileContent> loadFiles(final List<String> filepaths, final String fi
         return contents;
     }
 
+    private void validateFileSizes(final List<String> filePaths, final String fileContext, final String s3Storage) {
+        // Collect file sizes along with their paths
+        final Map<String, Long> fileSizesWithPaths = getFileSizesWithPaths(filePaths, fileContext, s3Storage);
+
+        // Filter files exceeding maximum size
+        final Map<String, Long> oversizedFiles = fileService.getOversizedFiles(fileSizesWithPaths);
+        // Handle oversized files
+        if (!oversizedFiles.isEmpty()) {
+            final String filesOverMaxString = oversizedFiles.entrySet().stream()
+                    .map(entry -> entry.getKey() + ": " + DataSize.ofBytes(entry.getValue()).toMegabytes() + " MB")
+                    .collect(Collectors.joining(System.lineSeparator()));
+            throw new BpmnError(FILE_SIZE_ERROR,
+                    String.format("The following files exceed the maximum size of %d MB:%n%s", fileService.getMaxFileSize().toMegabytes(), filesOverMaxString));
+        }
+
+        // Validate total batch size
+        final DataSize totalFileSize = fileService.getTotalBatchSize(fileSizesWithPaths);
+        if (!fileService.isValidBatchSize(totalFileSize))
+            throw new BpmnError(BATCH_SIZE_ERROR, String.format("Batch size of %d MB is too large. Allowed are %d MB.",
```

Comment:
Größe in MB wird aufgerundet. Bei geringer Überschreitung wird zweimal die gleiche Zahl in der Meldung angezeigt.